From b2dd005ab51cece1ae11f1ff664e424e7faf54f0 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 11:20:38 +0800
Subject: [PATCH 246/301] AOSCOS: drm: radeon: limit MIPS Loongson-3
 workarounds for Juniper

Guard MIPS Loongson-3 specific hacks around CONFIG_MACH_LOONGSON64.

Fixes: "AOSCOS: drm/radeon: Fix hibernation for JUNIPER on Loongson"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/radeon/evergreen.c | 2 ++
 drivers/gpu/drm/radeon/radeon_pm.c | 5 +++++
 2 files changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/radeon/evergreen.c b/drivers/gpu/drm/radeon/evergreen.c
index 493022119c1a..1a15f9264ead 100644
--- a/drivers/gpu/drm/radeon/evergreen.c
+++ b/drivers/gpu/drm/radeon/evergreen.c
@@ -1010,8 +1010,10 @@ static void evergreen_init_golden_registers(struct radeon_device *rdev)
 						 (const u32)ARRAY_SIZE(cypress_mgcg_init));
 		break;
 	case CHIP_JUNIPER:
+#ifdef CONFIG_MACH_LOONGSON64
 		if (rdev->pdev->subsystem_vendor == PCI_VENDOR_ID_ATI)
 			break; /* Fix hibernation for Loongson */
+#endif
 
 		radeon_program_register_sequence(rdev,
 						 evergreen_golden_registers,
diff --git a/drivers/gpu/drm/radeon/radeon_pm.c b/drivers/gpu/drm/radeon/radeon_pm.c
index c6c60f1b64cb..539065746db0 100644
--- a/drivers/gpu/drm/radeon/radeon_pm.c
+++ b/drivers/gpu/drm/radeon/radeon_pm.c
@@ -1520,7 +1520,9 @@ int radeon_pm_init(struct radeon_device *rdev)
 	case CHIP_RS780:
 	case CHIP_RS880:
 	case CHIP_RV770:
+#ifdef CONFIG_MACH_LOONGSON64
 	case CHIP_JUNIPER:
+#endif
 		/* DPM requires the RLC, RV770+ dGPU requires SMC */
 		if (!rdev->rlc_fw)
 			rdev->pm.pm_method = PM_METHOD_PROFILE;
@@ -1538,6 +1540,9 @@ int radeon_pm_init(struct radeon_device *rdev)
 	case CHIP_RV740:
 	case CHIP_CEDAR:
 	case CHIP_REDWOOD:
+#ifndef CONFIG_MACH_LOONGSON64
+	case CHIP_JUNIPER:
+#endif
 	case CHIP_CYPRESS:
 	case CHIP_HEMLOCK:
 	case CHIP_PALM:
-- 
2.48.1

